app/controllers/transfer_controller.rb

```diff
@@ -42,7 +44,14 @@ def select
   end
 
   def show
+    # Load the details of the requested Transfer record
     @transfer = Transfer.find(params[:id])
+
+    # Load the Thesis records for the period covered by this Transfer (the
+    # graduation month/year, and the department)
+    @theses = Thesis.all
+    @theses = @theses.where('grad_date == ?', @transfer.grad_date)
+    @theses = @theses.includes(:departments).where("departments.name_dw == ?", @transfer.department.name_dw).references(:departments)
```

Comment:
When reading this it feels like it'll be a super inefficient set of queries, but ActiveRecord turns this into a single query.

To relieve that cognitive confusion as to what is going on, it may be clearer to change it slightly while not changing the actual query ActiveRecord will compile to under the hood like:

```
@theses = Thesis.where('grad_date == ?', @transfer.grad_date)
@theses = @theses.includes(:departments).where("departments.name_dw == ?", @transfer.department.name_dw).references(:departments)
```

If that doesn't read clearer to you that we aren't _actually_ loading all the Thesis then no worries on changing it as I confirmed it makes absolutely no difference on the compiled db query.

TLDR: ActiveRecord is smarter than I give it credit for and my brain doesn't trust it so I'd like to be nice to my future brain but this is absolutely not a required change.

Reply:
Honestly, I wasn't sure whether this was going to be inefficient, or if a two-line approach (or even just chaining them all together into one expression) would be better. I'm happy to go with a two-line approach here - will push an updated commit in a bit.